research/2d-decoding/rectangle_reader.py

```diff
@@ -21,17 +31,21 @@ def main():
                     break
                 cv2.imshow("Square reader", frame)
                 k = cv2.waitKey(1)
-                if k%256 == 27:
+                if k % 256 == 27:
                     # ESC pressed
                     print("Escape hit, closing...")
                     break
-                cv2.imwrite("webcam.png", frame)
-                readImage("webcam.png")
+                if k % 256 == 32:
+                    # Space pressed
+                    cv2.imwrite("webcam.png", frame)
+                    print("Capture taken, processing...")
+                    process("webcam.png")
```

Comment:
Unnecessary io go brrrr

Reply:
Unnecessary, yes. A problem, no. Saving this frame takes a negligible amount of time and lets us easily string all the steps together to look pretty :). (It also means there is only one process case to deal with).